rosidl_cmake/cmake/rosidl_generate_interfaces.cmake

```diff
@@ -137,22 +137,50 @@ macro(rosidl_generate_interfaces target)
   # afterwards all remaining interface files are .idl files
   list(APPEND _idl_tuples ${_idl_adapter_tuples})
 
-  # to generate action interfaces, we need to depend on "action_msgs"
+  set(_service_msgs_added FALSE)
+  set(_action_msgs_added FALSE)
```

Comment:
instead of the new flags, can't you use `service_msgs_FOUND` and `action_msgs_FOUND` directly?

Reply:
They aren't declared until the first `find_package` so it would be a little messy. But we could definitely do that if requested.

Reply:
you don't need to declare variables in cmake

```cmake
if (service_msgs_FOUND)
  message("A")
else()
  message("B")
endif()
```

will print `B`, if service_msgs_FOUND is undefined or it's set to a false constant.
From [cmake docs](https://cmake.org/cmake/help/latest/command/if.html#basic-expressions):

> if(\<variable\>)
True if given a variable that is defined to a value that is not a false constant. False otherwise, including if the variable is undefined. Note that macro arguments are not variables. [Environment Variables](https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#cmake-language-environment-variables) also cannot be tested this way, e.g. if(ENV{some_var}) will always evaluate to false.

Reply:
I've refactored the loop, how's this? [91ea5a9](https://github.com/ros2/rosidl/pull/700/commits/91ea5a9062cecf44e8d77fc730c06ea89e922593)